src/GovUk.Education.ExploreEducationStatistics.Admin/Services/UserManagementService.cs

```diff
@@ -166,69 +174,78 @@ public async Task<UserViewModel> GetAsync(string userId)
             return null;
         }
 
-        public async Task<List<UserViewModel>> ListPendingAsync()
+        public async Task<Either<ActionResult, List<UserViewModel>>> ListPendingInvitesAsync()
         {
-            var pendingUsers = await _usersAndRolesDbContext.UserInvites.Where(u => u.Accepted == false)
-                .OrderBy(x => x.Email).Select(u => new UserViewModel
+            return await _userService
+                .CheckCanManageAllUsers()
+                .OnSuccess(async () =>
                 {
-                    Email = u.Email,
-                    Role = u.Role.Name
-                }).ToListAsync();
-
-            return pendingUsers;
+                    var pendingUsers = await _usersAndRolesDbContext.UserInvites.Where(u => u.Accepted == false)
```

Comment:
Can probably just put the return here rather than a one-use variable name - keeps the IDE happy :)

Reply:
In fact, can probably then shorten to:

```
        public async Task<Either<ActionResult, List<UserViewModel>>> ListPendingInvitesAsync()
        {
            return await _userService
                .CheckCanManageAllUsers()
                .OnSuccess(async () => 
                    await _usersAndRolesDbContext.UserInvites.Where(u => u.Accepted == false)
                        .OrderBy(x => x.Email).Select(u => new UserViewModel
                        {
                            Email = u.Email,
                            Role = u.Role.Name
                        }).ToListAsync()
                    );
        }
```

if you wanted to...